src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -67,54 +130,64 @@ void ZooKeeper::init(ZooKeeperArgs args_)
         if (args.hosts.empty())
             throw KeeperException::fromMessage(Coordination::Error::ZBADARGUMENTS, "No hosts passed to ZooKeeper constructor.");
 
-        Coordination::ZooKeeper::Nodes nodes;
-        nodes.reserve(args.hosts.size());
+        if (args.get_priority_load_balancing.load_balancing == DB::LoadBalancing::KEEPER_LOCAL_AVAILABILITY_ZONE)
+        {
+            tryConnectSameAZKeeper();
+        }
+        // TODO: reduce nesting level.
+        else
+        {
+            Coordination::ZooKeeper::Nodes nodes;
+            nodes.reserve(args.hosts.size());
 
-        /// Shuffle the hosts to distribute the load among ZooKeeper nodes.
-        std::vector<ShuffleHost> shuffled_hosts = shuffleHosts();
+            /// Shuffle the hosts to distribute the load among ZooKeeper nodes.
+            std::vector<ShuffleHost> shuffled_hosts;
+            shuffled_hosts = shuffleHosts();
 
-        bool dns_error = false;
-        for (auto & host : shuffled_hosts)
-        {
-            auto & host_string = host.host;
-            try
+            // TODO: add DNS check as well.
```

Comment:
Will restore the DNS logic below and incorporate that to local AZ flow as well..

Reply:
Actually not much diff, Github is confused with indentation.